sormas-app/app/src/main/java/de/symeda/sormas/app/component/dialog/SynchronizationDialog.java

```diff
@@ -0,0 +1,610 @@
+				logger.error("InterruptedException when trying to perform BINDING_LOCK.wait() in SynchronizationDialog.updateSynchronizationStep: " + e.getMessage());
+			}
+		}
+
+		updatingProgressItemBindings = true;
+		currentBinding = null;
+		currentSyncStep = synchronizationStep;
+
+		switch (synchronizationStep) {
+		case PUSH_NEW:
+			showPushNewProgressItems();
+			break;
+		case PULL_INFRASTRUCTURE:
+			showPullInfrastructureProgressItems(false);
+			break;
+		case PULL_INFRASTRUCTURE_INITIAL:
+			showPullInfrastructureProgressItems(true);
+			break;
+		case CLEAR_UP:
+			showClearUpProgressItems(true);
+			break;
+		case SYNCHRONIZE:
+			showSynchronizeProgressItems();
+			break;
+		case CLEAR_UP_INFRASTRUCTURE:
+			showClearUpInfrastructureProgressItems(true);
+			break;
+		case DELETE_OBSOLETE:
+			showDeleteObsoleteProgressItems();
+			break;
+		case PULL_MODIFIED:
+			showPullModifiedProgressItems();
+			break;
+		case REPULL:
+			showRepullProgressItems();
+			break;
+		default:
+			throw new IllegalArgumentException(synchronizationStep.name());
+		}
+
+		getActivity().runOnUiThread(() -> {
+			contentBinding.stepsLayout.setCurrentStep(++currentSyncStepIndex);
+			contentBinding.stepsLayout.stepText
+				.setText(String.format(getActivity().getResources().getString(synchronizationStep.getCaptionResource()), currentSyncStepIndex));
+		});
+	}
+
+
+	private void waitForProgressItemBindings() {
+
+		if (!updatingProgressItemBindings && currentBinding != null) {
+			return;
+		}
+
+		synchronized (BINDING_LOCK) {
+			try {
+				if (updatingProgressItemBindings || currentBinding == null) {
+					BINDING_LOCK.wait();
+				}
+			} catch (InterruptedException e) {
+				logger.error("InterruptedException when trying to perform BINDING_LOCK.wait() in SynchronizationDialog.waitForProgressItemBindings: " + e.getMessage());
+			}
+		}
+	}
+
+	private void updateStepNumber(int stepNumber) {
+		getActivity().runOnUiThread(() -> contentBinding.stepsLayout.setStepNumber(stepNumber));
+	}
+
+	private void loadNext() {
+
+		waitForProgressItemBindings();
+
+		currentBinding.setActive(false);
+		currentBinding.setDone(true);
+
+		if (progressItemBindings.size() > ++progressItemBindingsIndex) {
+			currentBinding = progressItemBindings.get(progressItemBindingsIndex);
+			currentBinding.setActive(true);
+		}
+	}
+
+	private void updatePulls(int pulled) {
+		waitForProgressItemBindings();
+		currentBinding.setPulls(currentBinding.getPulls() + pulled);
+	}
+
+	private void updatePushes(int pushed) {
+		waitForProgressItemBindings();
+		currentBinding.setPushes(currentBinding.getPushes() + pushed);
+	}
+
+	private void updatePushTotal(int pushTotal) {
+		waitForProgressItemBindings();
+		currentBinding.setPushTotal(pushTotal);
+	}
+
+	private void updateDeletions(int deleted) {
+		waitForProgressItemBindings();
+		currentBinding.setDeletions(currentBinding.getDeletions() + deleted);
+	}
+
+	private void showNextCleanupItems() {
```

Comment:
I'd like to get rid of this method. It splits the updatingProgressItemBindings logic into a second "branch" and I also didn't get why it's needed in the first place.

As far as I understand why could replace it by adding another two entries in SynchronizationStep enum (e.g. CLEAR_UP_INFRASTRUCTURE_MISSING and CLEAR_UP_MISSING) and calling updateSynchronizationStep instead.